File: java/lang/security/audit/crypto/weak-rsa.java
import java.security.KeyPairGenerator;

public class WeakRSA {

  static void rsaWeak() {
    // ruleid: use-of-weak-rsa-key
    KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA");
    keyGen.initialize(512);
  }

  static void rsaWeak_lowercase() {
    // ruleid: use-of-weak-rsa-key
    KeyPairGenerator keyGen = KeyPairGenerator.getInstance("rsa");
    keyGen.initialize(512);
  }

  static void rsaOK() {
    // ok: use-of-weak-rsa-key
    KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA");
    keyGen.initialize(2048);
  }
}

File: kotlin/lang/security/weak-rsa.kt
import java.security.KeyPairGenerator

public class WeakRSA {

  fun rsaWeak(): Void {
    // ruleid: use-of-weak-rsa-key
    val keyGen: KeyPairGenerator = KeyPairGenerator.getInstance("RSA")
    keyGen.initialize(512)
  }

  fun rsaOK(): Void {
    // ok: use-of-weak-rsa-key
    val keyGen = KeyPairGenerator.getInstance("RSA");
    keyGen.initialize(2048);
  }
}
